hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -828,49 +841,40 @@ async def create(billing_project: str,
         if not deploy_config:
             deploy_config = get_deploy_config()
         url = deploy_config.base_url('batch')
-        if session is None:
-            session = httpx.client_session()
         if headers is None:
             headers = {}
-        if _token:
-            headers['Authorization'] = f'Bearer {_token}'
+        credentials: CloudCredentials
+        if _token is not None:
+            credentials = HailExplicitTokenCredentials(_token)
         else:
-            headers.update(service_auth_headers(deploy_config, 'batch', token_file=token_file))
+            credentials = hail_credentials(credentials_file=token_file)
         return BatchClient(
             billing_project=billing_project,
             url=url,
-            session=session,
+            session=Session(credentials=credentials, http_session=session),
             headers=headers)
 
     def __init__(self,
                  billing_project: str,
                  url: str,
-                 session: httpx.ClientSession,
+                 session: Session,
                  headers: Dict[str, str]):
         self.billing_project = billing_project
         self.url = url
-        self._session: Optional[httpx.ClientSession] = session
+        self._session: Session = session
         self._headers = headers
 
-    async def _get(self, path, params=None) -> aiohttp.client_reqrep.ClientResponse:
-        assert self._session
-        return await request_retry_transient_errors(
-            self._session, 'GET', self.url + path, params=params, headers=self._headers
-        )
+    async def _get(self, path, params=None) -> aiohttp.ClientResponse:
```

Comment:
The return type here looks right from the code in main, but I thought we would have returned httpx.ClientResponse.

Reply:
hm ya I'll fix these types, good call

Reply:
Actually, weird, I can't find a single usage of `httpx.ClientResponse` in the codebase. Is that dead code? Either way it looks like this is the correct type from what I can see in `httpx.ClientSession._request`